app/Models/Build.php

```diff
@@ -235,4 +235,9 @@ public function displayVersion()
     {
         return preg_replace('#[^0-9.]#', '', $this->version);
     }
+
+    public static function onlineUsers()
+    {
+        return static::where('allow_bancho', 1)->sum('users');
```

Comment:
I'm not sure this is correct. I think it'll double-count lazer users.

The reason for that is that lazer build tracking is a little wonky. See [this discord message](https://discord.com/channels/188630481301012481/188630652340404224/1184782370105200753) for an example, but in short, for every _actual_ lazer release:

- there will be a "main" lazer release row with the actual version, with `stream_id` set correctly to 7, but with no hash,
- and there will be "fake" lazer release rows, one per platform, with `null` `stream_id` but hashes set to match the appropriate platform.

And notably all of them will have `allow_bancho` set to 1.

I'd recommend looking through https://github.com/ppy/osu-server-spectator/pull/204 to see how `osu-server-spectator` handles that right now. Maybe we want to clean this up at some point (maybe even now), but I'd rather point this out now before this PR goes too far.

Reply:
Ah, thanks for that. I've updated to add an extra `where` that ensures `hash` is `null`. I thought about adding an extra `where` to ensure that `stream_id` is `7` however I'm unsure if this is correct? In the PR linked, the tests are using any build ids while `GetAllMainLazerBuildsAsync` is doing `WHERE stream_id = 7` so would appreciate some clarification there.

Reply:
I'd say that basing it on stream not being null is more likely correct than on null hash. And I'm not sure how the build counts look when it comes to stable releases. Probably needs @peppy double-check.

Reply:
That makes more sense. I've applied that, but will keep this open awaiting a double-check from peppy regarding stable.

Reply:
I don't think the main lazer build will ever get a hash, but either sounds fine for now.

Reply:
Sounds good, any concerns regarding stable?